pkg/api/controllers/samlconfig/samlconfig.go

```diff
@@ -0,0 +1,82 @@
+package samlconfig
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rancher/rancher/pkg/auth/providers/saml"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type authProvider struct {
+	authConfigs v3.AuthConfigInterface
+}
+
+func Register(apiContext *config.ScaledContext) {
+	a := newAuthProvider(apiContext)
+	apiContext.Management.AuthConfigs("").AddHandler("authConfigController", a.sync)
+}
+
+func newAuthProvider(apiContext *config.ScaledContext) *authProvider {
+	a := &authProvider{
+		authConfigs: apiContext.Management.AuthConfigs(""),
+	}
+	return a
+}
+
+func (a *authProvider) sync(key string, config *v3.AuthConfig) error {
```

Comment:
This sync method is potentially called a lot, even when there has not been a change to the config. We should add some intelligene here or in `InitializeSamlServiceProvider`so that we are smart about when we reinitialize